docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,226 @@
+1. The same session from a previous app run is still ongoing. No action is required
+2. Another session is ongoing in which case a flush ping is required to submit data from a previous session and start recording data for the new session
+3. No session is ongoing but there is a lingering session id indicating that there is still lingering data from a previous session in storage and a flush ping is required
+4. No session is ongoing and no lingering session id is in storage. No action is required
+
+#### DAEMON
+
+On the daemon there are only two possibilities:
+
+1. No session is ongoing but there is a lingering session id indicating that there is still lingering data from a previous session in storage and a flush ping is required
+2. No session is ongoing and no lingering session id is in storage. No action is required
+
+## APPENDIX
+
+### APPENDIX 1: PSEUDO CODE
+
+#### Main application
+
+```cpp
+MainApp::init() {
+  String lingeringSessionId = LocalStorage::get(SESSION_ID_STORAGE_KEY);
+  // If this is not empty, the VPN was not deactivated from the app.
+  if (!lingeringSessionId.isEmpty()) {
+    if (Daemon::vpnIsOn()) {
+      try {
+        String sessionId = Daemon::getSessionId();
+        // Check if we are still in the same session
+        if (sessionId != lingeringSessionId) {
+          // Flush any lingering data from a previous session
+          pings::main_app::session.submit(INIT_FLUSH_REASON);
+
+          metrics::main_app::sessionId.set(sessionId);
+          // Record session::* context metrics e.g. app_exclusions.
+
+          LocalStorage::set(SESSION_ID_STORAGE_KEY, sessionId);
+        }
+
+        // Schedule the timer.
+        MainApp::schedulePingTimer();
+      } catch {
+        // Note: If communication with the daemon doesn't work,
+        // let's assume the VPN is off. This is very unlikely, given we
+        // have just communicated with the daemon through Daemon::vpnIsOn();
+
+        // Flush any lingering data from a previous session
+        pings::main_app::session.submit(INIT_FLUSH_REASON);
+        // Clear the sessionId storage
+        LocalStorage::clear(SESSION_ID_STORAGE_KEY);
+      }
+    } else {
+      // Flush any lingering data from a previous session
+      pings::main_app::session.submit(INIT_FLUSH_REASON);
+      // Clear the sessionId storage
+      LocalStorage::clear(SESSION_ID_STORAGE_KEY);
+    }
+  }
+}
+
+MainApp::onActivate() {
+  // Let's flush any lingering data for before activating the VPN.
+  // We want the active session pings to only include data from the current session.
+  //
+  // This ping should be always empty. This is a way to monitor if it isn't
+  pings::main_app::session.submit(FLUSH_REASON);
+
+  String sessionId = generateUUID();
+  try {
+    Daemon::activate(config, sessionId);
+    // Note: session::* metrics should only be recorded from now on,
+    // so that flush pings don't contain unexpected data.
+
+    metrics::main_app::sessionId.set(sessionId);
```

Comment:
Nit: Any reason you decided not to set this metric before activating the daemon (line 134)? If we inverted, we could just pull the UUID from the metric creation.

Reply:
Not sure, probably no reason.